### ISSUE TITLE
Turning off indexing, Pylance gives different results for completions (2023.12.1 did not)

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#5397]
Create code like so after installing debugpy in your venv:
```python
import debugpy._vendored.pydevd._pydevd_bundle._debug_adapter.pydevd_schema
x = debugpy._vendored.pydevd._pydevd_bundle._debug_adapter.pydevd_schema.BaseSchema()
x.
```

1. Turn on indexing: `"python.analysis.indexing": true`
2. Stick the cursor after the `x.`
3. Bring up completions
4. Turn off indexing
5. Restart VS code
6. Bring up completions again

Completions are different after turning off indexing. 

In 2023.12.1, they remain the same.

**Comment by rchiodo** [comment]
Found while testing: https://github.com/microsoft/pyrx/issues/4538

**Comment by rchiodo** [comment]
Did some digging, it looks like at this line here:

```typescript
            // The scope for this class becomes the "fields" for the corresponding type.
            const innerScope = ScopeUtils.getScopeForNode(node.suite);
            classType.details.fields = innerScope?.symbolTable
                ? new Map<string, Symbol>(innerScope.symbolTable)
                : new Map<string, Symbol>();
```

The innerScope doesn't have the same members. Not sure why yet.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2024.4.101, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20244101-26-January-2024-PreRelease)
